### PR TITLE
Add documentation for `RescriptReactRouter`

### PR DESCRIPTION
**Pull request by diogomqbm**
I really missed this on documentation so I decided to contribute by adding it with a basic example. Got some texts from [ReasonReact Router](https://reasonml.github.io/reason-react/docs/en/router) old docs and wrote the example myself.

### CODE REVIEW

File: pages/docs/react/latest/router.mdx [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+title: Router
+description: "Basic concepts for navigation and routing in ReScript & React"
+canonical: "/docs/react/latest/router"
+---
+
+# Router
+
+<Intro>
```
Comment (ryyppy) [inline pages/docs/react/latest/router.mdx:9]:
```suggestion
```

Intro with list items looks broken

File: pages/docs/react/latest/router.mdx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+RescriptReact comes with a router! We've leveraged the language and library features in order to create a router that's:
+
+- The simplest, thinnest possible.
+- Easily pluggable anywhere into your existing code.
+- Performant and tiny.
+
+</Intro>
```
Comment (ryyppy) [inline pages/docs/react/latest/router.mdx:17]:
```suggestion
```

**Review by ryyppy — CHANGES_REQUESTED** [review]
Thanks for your contribution! At some point I wanted to adapt the Router docs to reflect the same structure and style as the rest of the docs, but for now taking the original version sounds good to me.

After addressing the few changes, I'd be happy to merge.

**Comment by diogomqbm** [comment]
Done! Thanks!

**Pull request merged**